astropy/coordinates/sky_coordinate.py

```diff
@@ -1147,14 +1150,19 @@ def _get_units(args, kwargs):
     return units
 
 
-def _parse_coordinate_arg(coords, frame, units):
+def _parse_coordinate_arg(coords, frame, units, scinit_kwargs):
```

Comment:
I feel like just `init_kwargs` or even just `kwargs` or `options` would be fine names for this.